packages/microblog/src/client.ts

```diff
@@ -0,0 +1,284 @@
+    description?: string,
+    hardcode: Partial<Profile.Record> = {},
+  ): Promise<AdxUri> {
+    const uri = await this.repo
+      .collection('bsky/profile')
+      .create('blueskyweb.xyz:Profile', {
+        $type: 'blueskyweb.xyz:Profile',
+        displayName,
+        description,
+        ...hardcode,
+      })
+    return uri
+  }
+
+  async createPost(
+    text: string,
+    entities?: Post.Entity[],
+    hardcode: Partial<Post.Record> = {},
+  ): Promise<AdxUri> {
+    const uri = await this.repo
+      .collection('bsky/posts')
+      .create('blueskyweb.xyz:Post', {
+        $type: 'blueskyweb.xyz:Post',
+        text,
+        entities,
+        createdAt: new Date().toISOString(),
+        ...hardcode,
+      })
+    return uri
+  }
+
+  async reply(
+    root: AdxUri,
+    parent: AdxUri,
+    text: string,
+    hardcode: Partial<Post.Record> = {},
+  ): Promise<AdxUri> {
+    const uri = await this.repo
+      .collection('bsky/posts')
+      .create('blueskyweb.xyz:Post', {
+        $type: 'blueskyweb.xyz:Post',
+        text,
+        reply: {
+          root: root.toString(),
+          parent: parent.toString(),
+        },
+        createdAt: new Date().toISOString(),
+        ...hardcode,
+      })
+    return uri
+  }
+
+  async likePost(
+    uri: AdxUri,
+    hardcode: Partial<Like.Record> = {},
+  ): Promise<AdxUri> {
+    const likeUri = await this.repo
+      .collection('bsky/likes')
+      .create('blueskyweb.xyz:Like', {
+        $type: 'blueskyweb.xyz:Like',
+        subject: uri.toString(),
+        createdAt: new Date().toISOString(),
+        ...hardcode,
+      })
+    return likeUri
+  }
+
+  async repost(
+    uri: AdxUri,
+    hardcode: Partial<Repost.Record> = {},
+  ): Promise<AdxUri> {
+    const repostUri = await this.repo
+      .collection('bsky/reposts')
+      .create('blueskyweb.xyz:Repost', {
+        $type: 'blueskyweb.xyz:Repost',
+        subject: uri.toString(),
+        createdAt: new Date().toISOString(),
+        ...hardcode,
+      })
+    return repostUri
+  }
+
+  async followUser(
+    did: string,
+    hardcode: Partial<Follow.Record> = {},
+  ): Promise<AdxUri> {
+    const uri = await this.repo
+      .collection('bsky/follows')
+      .create('blueskyweb.xyz:Follow', {
+        $type: 'blueskyweb.xyz:Follow',
+        subject: did,
+        createdAt: new Date().toISOString(),
+        ...hardcode,
+      })
+    return uri
+  }
+
+  async giveBadge(
+    did: string,
+    type: string,
+    tag?: string,
+    hardcode: Partial<Badge.Record> = {},
+  ): Promise<AdxUri> {
+    const uri = await this.repo
+      .collection('bsky/badges')
+      .create('blueskyweb.xyz:Badge', {
+        $type: 'blueskyweb.xyz:Badge',
+        assertion: {
+          type,
+          tag,
+        },
+        subject: did,
+        createdAt: new Date().toISOString(),
+        ...hardcode,
+      })
+    return uri
+  }
+
+  async acceptBadge(badgeUri: AdxUri): Promise<AdxUri> {
```

Comment:
This is the profile mutation that would probs make sense as a procedure 🤔

Reply:
Yeah makes sense